.github/workflows/generate_graph_test.yml

```
@@ -0,0 +1,81 @@
+name: Build and test a Kiota generated SDK with packages
+on:
+  workflow_dispatch:
+    inputs:
+      KIOTA_BRANCH:
+        description: 'Branch to build kiota from'
+        type: string
+        default: 'nikithauc/model-as-interfaces'
+      KIOTA_REPO:
+        description: 'Repo to build kiota from'
+        type: string
+        default: 'https://github.com/microsoft/kiota.git'
+      OPEN_API_DEF:
+        description: 'OpenAPI definition to build test SDK for'
+        type: string
+        default: 'https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/openapi.yaml'
+
+  push:
+    branches: [ nikithauc/model-interface-serialization ]
+  pull_request:
+
+env:
+  KIOTA_BRANCH: 'nikithauc/model-as-interfaces'
```

```suggestion
  KIOTA_BRANCH: ${{ inputs.KIOTA_BRANCH }}
```

can you define it this way instead? so you don't have to read both in the workflow?

---

I was finding that when it was triggered from a PR the inputs were not set so I added the env as a fallback.